pkg/cni/types/types.go

```diff
@@ -196,12 +196,27 @@ func (cl *confLoader) LoadConf(args *skel.CmdArgs) (*NetConf, error) {
 
 	if n.Args != nil && n.Args.ArgsCNI != nil {
 		n.IPAM.Features.AllocateDefaultGateway = n.Args.ArgsCNI.AllocateDefaultGateway
+		if n.Args.ArgsCNI.AllocateIPWithIndex != nil && *n.Args.ArgsCNI.AllocateIPWithIndex < 0 {
+			return nil, fmt.Errorf("allocateIPWithIndex can't be negative")
+		}
 		n.IPAM.Features.AllocateIPWithIndex = n.Args.ArgsCNI.AllocateIPWithIndex
 	}
 
 	if n.IPAM.Features.AllocateDefaultGateway {
 		if len(n.IPAM.RequestedIPs) > 0 {
-			return nil, fmt.Errorf("allocatedDefaultGateway can't be used together with static IP request")
+			return nil, fmt.Errorf("allocateDefaultGateway can't be used together with static IP request")
+		}
+		if n.IPAM.Features.AllocateIPWithIndex != nil {
+			return nil, fmt.Errorf("allocateDefaultGateway can't be used together allocateIPWithIndex")
+		}
+	}
+
+	if n.IPAM.Features.AllocateIPWithIndex != nil {
+		if len(n.IPAM.RequestedIPs) > 0 {
+			return nil, fmt.Errorf("allocateIPWithIndex can't be used together with static IP request")
+		}
+		if n.IPAM.Features.AllocateDefaultGateway {
```

Comment:
It looks like we will never reach this code. This condition is already handled on line 209